raiden/storage/wal.py

```diff
@@ -66,19 +59,20 @@ def restore_to_state_change(
 ST = TypeVar("ST", bound=State)
 
 
-class SavedState(NamedTuple):
+@dataclass(frozen=True)
+class SavedState(Generic[ST]):
```

Comment:
typo in the docstring not introduced by this PR but well you can fix it: `Saves the state the id of the state change` -> `Saves the state **and** the id of the state change`